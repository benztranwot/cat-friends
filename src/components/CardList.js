import React from "react";
import Card from "./Card";

const CardList = ({ cats }) => {
  // if (true) {
  //   throw new Error("Test Error");
  // }

  return (
    <div>
      {cats.map((cat) => (
        <Card key={cat.id} id={cat.id} name={cat.name} email={cat.email} username={cat.username} />
      ))}
    </div>
  );
};

export default CardList;
