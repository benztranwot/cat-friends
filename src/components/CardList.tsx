import Card from "./Card";
import { Cat } from "../containers/App";

const CardList = ({ cats }: { cats: Array<Cat> }) => {
  // if (true) {
  //   throw new Error("Test Error");
  // }

  return (
    <div>
      {cats.map((cat) => (
        <Card key={cat.id} id={cat.id} name={cat.name} email={cat.email} username={cat.username} />
      ))}
    </div>
  );
};

export default CardList;
