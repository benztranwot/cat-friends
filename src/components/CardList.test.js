import renderer from "react-test-renderer";
import CardList from "./CardList";

test("expect to render Card component", () => {
  const mockCats = [
    {
      id: 1,
      name: "John Snow",
      email: "winter@is.coming",
      username: "JohnNorth",
    },
  ];
  const tree = renderer.create(<CardList cats={mockCats} />).toJSON();
  expect(tree).toMatchSnapshot();
});
