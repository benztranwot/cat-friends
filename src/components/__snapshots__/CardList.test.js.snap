// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card component 1`] = `
<div>
  <div
    className="tc bg-light-green dib br3 pa3 ma2 grow bw2 shadow-5"
    data-testid="card"
  >
    <img
      alt="cat"
      src="https://robohash.org/JohnNorth?set=set4"
    />
    <div>
      <h2>
        John Snow
      </h2>
      <p>
        winter@is.coming
      </p>
    </div>
  </div>
</div>
`;
